examples/moderation_example.py

```diff
@@ -21,51 +21,69 @@
 
 import lightbulb
 
-bot = lightbulb.BotApp(token="YOUR_TOKEN", intents=hikari.Intents.ALL_UNPRIVILEGED)
+bot = hikari.GatewayBot(token="...")
+client = lightbulb.client_from_app(bot)
 
+bot.subscribe(hikari.StartingEvent, client.start)
 
-@bot.command()
-@lightbulb.option("reason", "Reason for the ban", required=False)
-@lightbulb.option("user", "The user to ban.", type=hikari.User)
-@lightbulb.command("ban", "Ban a user from the server.")
-@lightbulb.implements(lightbulb.SlashCommand)
-async def ban(ctx: lightbulb.SlashContext) -> None:
-    """Ban a user from the server with an optional reason."""
-    if not ctx.guild_id:
-        await ctx.respond("This command can only be used in a guild.")
-        return
 
-    # Create a deferred response as the ban may take longer than 3 seconds
-    await ctx.respond(hikari.ResponseType.DEFERRED_MESSAGE_CREATE)
-    # Perform the ban
-    await ctx.app.rest.ban_user(ctx.guild_id, ctx.options.user.id, reason=ctx.options.reason or hikari.UNDEFINED)
-    # Provide feedback to the moderator
-    await ctx.respond(f"Banned {ctx.options.user.mention}.\n**Reason:** {ctx.options.reason or 'No reason provided.'}")
+@client.register()
+class Ban(
+    lightbulb.SlashCommand,
+    name="ban",
+    description="Bans a user from the server",
+):
+    # Order of options go from top to bottom
+    user = lightbulb.user("user", "The user to ban")
+    # Give non-required options a default value (e.g. default=None)
+    # Non-required options MUST appear after required options
+    # Required options do not have a default value
+    reason = lightbulb.string("reason", "Reason for the ban", default=None)
 
+    @lightbulb.invoke
+    async def invoke(self, ctx: lightbulb.Context) -> None:
+        """Ban a user from the server with an optional reason"""
+        if not ctx.guild_id:
+            await ctx.respond("This command can only be used in a guild.")
+            return
 
-@bot.command()
-@lightbulb.option("count", "The amount of messages to purge.", type=int, max_value=100, min_value=1)
-# You may also use pass_options to pass the options directly to the function
-@lightbulb.command("purge", "Purge a certain amount of messages from a channel.", pass_options=True)
-@lightbulb.implements(lightbulb.SlashCommand)
-async def purge(ctx: lightbulb.SlashContext, count: int) -> None:
-    """Purge a certain amount of messages from a channel."""
-    if not ctx.guild_id:
-        await ctx.respond("This command can only be used in a server.")
-        return
+        # Create a deferred response as the ban may take longer than 3 seconds
+        await ctx.respond(hikari.ResponseType.DEFERRED_MESSAGE_CREATE)
+        # Perform the ban
+        await ctx.app.rest.ban_user(ctx.guild_id, self.user.id, reason=self.reason or hikari.UNDEFINED)
```

Comment:
I don't think this will work as `Context` doesn't have an `app` field anymore - I would prefer to dependency-inject the `RESTClient` (using the type-hint `hikari.api.RESTClient` to the invoke function and use that directly. Same for the purge command.